azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/features/ResourceGroupApi.java

```diff
@@ -0,0 +1,99 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecomputearm.features;
+import java.io.Closeable;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Named;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+
+import org.jclouds.Fallbacks.EmptyListOnNotFoundOr404;
+import org.jclouds.Fallbacks.NullOnNotFoundOr404;
+import org.jclouds.Fallbacks.VoidOnNotFoundOr404;
+import org.jclouds.azurecomputearm.domain.ResourceGroup;
+import org.jclouds.azurecomputearm.functions.StatusCodeParser;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.oauth.v2.filters.OAuthFilter;
+import org.jclouds.rest.annotations.QueryParams;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.SelectJson;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.MapBinder;
+import org.jclouds.rest.annotations.PayloadParam;
+import org.jclouds.rest.annotations.PATCH;
+import org.jclouds.rest.annotations.ResponseParser;
+
+
+import org.jclouds.rest.binders.BindToJsonPayload;
+
+/**
+ * The Azure Resource Manager API includes operations for managing resource groups in your subscription.
+ *
+ * @see <a href="https://msdn.microsoft.com/en-us/library/azure/dn790546.aspx">docs</a>
+ */
+@Path("/subscriptions/{subscriptionid}")
+
+@QueryParams(keys = "api-version", values = "2015-01-01")
+@RequestFilters(OAuthFilter.class)
+@Consumes(MediaType.APPLICATION_JSON)
+public interface ResourceGroupApi extends Closeable{
+
+   @Named("resourcegroup:list")
+   @Path("resourcegroups")
+   @SelectJson("value")
+   @GET
+   @Fallback(EmptyListOnNotFoundOr404.class)
+   List<ResourceGroup> list();
+
+   @Named("resourcegroup:create")
+   @PUT
+   @Path("resourcegroups/{name}")
+   @Produces(MediaType.APPLICATION_JSON)
+   @MapBinder(BindToJsonPayload.class)
+   ResourceGroup create(@PathParam("name") String name, @PayloadParam("location") String location, @PayloadParam("tags")Map<String, String> tags);
+
+   @Named("resourcegroup:get")
+   @GET
+   @Path("resourcegroups/{name}")
+   @Fallback(NullOnNotFoundOr404.class)
+   @Nullable
+   ResourceGroup get(@PathParam("name") String name);
+
+   @Named("resourcegroup:update")
+   @PATCH
+   @Produces(MediaType.APPLICATION_JSON)
+   @Path("resourcegroups/{name}")
+   @Fallback(NullOnNotFoundOr404.class)
+   @Nullable
+   @MapBinder(BindToJsonPayload.class)
+   ResourceGroup update(@PathParam("name") String name, @PayloadParam("tags")Map<String, String> tags);
+
+   @Named("resourcegroup:delete")
+   @DELETE
+   @ResponseParser(StatusCodeParser.class)
```

Comment:
I see, in that case, I think it should be better to return something that already allow users to track the status. What about:

* Changing the method to return a URI.
* Change (and rename) the response parser so it:
 * Returns `null` if the operation succeeds and there is no task to track (response code 200).
 * Returns an URI object if there is an ongoing task (202 response code) so users can track it.
 * Propagates an exception otherwise.

This would facilitate tracking of ongoing jobs. Then we could also add a generic method to the root `AzureApi` (or create a JobApi or whatever) like:
```java
@ResponseParser(ParseJobStatus.class)
JobStatus jobDone(URI jobURI);
```

Where `JobStatus` could be a custom enum with `DONE, IN_PRORGESS, FAILED` states, and the response parser could return an appropriate value depending on the response code (200, 202, others).

This would help users keep track of async requests, and also will facilitate the implementation of the ComputeService if it needs to use such calls. WDYT @ritazh?